stract: 0-unstable-2024-09-14 -> 0-unstable-2024-12-11

eljamm opened the pull request:
The previous version had the wrong hash, thus the wrong package was being built.

Fixes: https://github.com/NixOS/nixpkgs/pull/376086

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eljamm committed:
stract: 0-unstable-2024-09-14 -> 0-unstable-2024-12-11

In the previous version, a fixed-output derivation in the source hash
was causing another package to be built instead of stract.

This commit updates stract to the latest version, fixing the source and
the build of the actual package.

eljamm committed:
stract: add ngi team to maintainers

eljamm commented:
This currently fails to build with:

```
error: failed to run custom build command for `utoipa-swagger-ui v8.0.3`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/utoipa-swagger-ui-b244f3caaea01521/build-script-build` (exit status: 101)
  --- stdout
  OUT_DIR: /build/source/target/x86_64-unknown-linux-gnu/release/build/utoipa-swagger-ui-eda506390aa385cb/out
  SWAGGER_UI_DOWNLOAD_URL: file:///nix/store/s6w503szg7zk40zm12wdfc7m1d1a2hkx-v5.17.14.zip
  cargo:rerun-if-changed="/nix/store/s6w503szg7zk40zm12wdfc7m1d1a2hkx-v5.17.14.zip"
  start copy to : "/build/source/target/x86_64-unknown-linux-gnu/release/build/utoipa-swagger-ui-eda506390aa385cb/out/s6w503szg7zk40zm12wdfc7m1d1a2hkx-v5.17.14.zip"

  --- stderr
  thread 'main' panicked at /build/stract-vendor/utoipa-swagger-ui-8.0.3/build.rs:168:47:
  called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 376162`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>stract</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>stract</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>stract</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>stract</li>
  </ul>
</details>

eljamm commented:
Seems to be the correct package this time 😁

```
Usage: stract <COMMAND>

Commands:
  indexer               Build an index
  centrality            Calculate centrality metrics that estimates a websites importance. These metrics are used to rank search results
  webgraph              Webgraph specific commands
  search-server         Deploy the search server
  entity-search-server  Deploy the entity search server
  api                   Deploy the json http api. The api interacts with the search servers, webgraph servers etc. to provide the necesarry functionality
  crawler               Deploy the crawler
  safety-classifier     Train or run inference on the classifier that predicts if a webpage is NSFW or SFW
  configure             Setup dev environment
  live-index            Commands for the live index
  web-spell             Build spell correction model
  site-stats            Compute statistics for sites
  ampc                  Commands to compute distributed graph algorithms
  admin                 Commands for the admin interface to manage stract
  help                  Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
```

alyssais commented on pkgs/by-name/st/stract/package.nix:
```diff
@@ -1,36 +1,71 @@
   buildInputs = [
+    oniguruma
     openssl
     curl
```
Did you check these two are still necessary?

eljamm replied:
openssl is still necessary, but curl isn't. I removed it from here and the nativeBuildInputs.

alyssais commented on pkgs/by-name/st/stract/package.nix:
```diff
@@ -1,36 +1,71 @@
+    rm -rf target/${stdenv.hostPlatform.config}/release/build/
+  '';
+
   meta = {
```
(Optional) You could add an unstableGitUpdater here

alyssais commented on pkgs/by-name/st/stract/package.nix:
```diff
@@ -1,36 +1,71 @@
   cargoDepsName = "stract";
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [
+    curl
```
What's native curl used for?

alyssais replied:
Also looks like it's added in the wrong commit — it's in the formatting one.

eljamm replied:
Don't know how that got there, thanks.

eljamm committed:
stract: add update script

alyssais commented:
Would be good if the first commit message mentioned something about how it wasn't actually a stract package before, and if the third explained why the change was being made.

eljamm commented on pkgs/by-name/st/stract/package.nix:
```diff
@@ -1,36 +1,75 @@
+
   meta = {
-    description = "Open source web search engine hosted at stract.com targeted towards tinkerers and developers.";
+    description = "Open source web search engine targeted towards tinkerers and developers";
```
>  and if the third explained why the change was being made.

I've opted to removing this because I think it makes the description clearer and because this can realistically be hosted anywhere.

What do you think @alyssais? Should it stay the same?

ailsa-sun replied:
Hi there! I was the original author of this commit, and hadn't gotten to it until now. Is there anything I should do to fix it at this moment?

eljamm replied:
Hey @ailsa-sun! Packaging-wise I think this is fine, now. What you could do is test that the package works as intended if it's not too difficult and if you have the time.

eljamm replied:
> I've opted to removing this because I think it makes the description clearer and because this can realistically be hosted anywhere.

I've restored the original description (minus the period). I thought about moving some of it into a `longDescription`, but I guess there is no need to complicate things now as the original goal was to fix the build.

alyssais replied:
That's great, thanks!

eljamm committed:
stract: update meta.description

Remove trailing period and move information to a long description for
better clarity.

alyssais commented on pkgs/by-name/st/stract/package.nix:
```diff
@@ -1,36 +1,79 @@
+  # swagger-ui will once more be copied in the target directory during the check phase
+  # Not deleting the existing unpacked archive leads to a `PermissionDenied` error
+  preCheck = ''
+    rm -rf target/${stdenv.hostPlatform.config}/release/build/
```
This won't always be correct.

```suggestion
    rm -rf target/${stdenv.hostPlatform.rust.cargoShortTarget}/release/build/
```
